src/data/sparse_page_source.h

```diff
@@ -20,6 +20,12 @@
 
 namespace xgboost {
 namespace data {
+
+class PageSource {
+ public:
+  virtual ~PageSource() DMLC_THROW_EXCEPTION {}
```

Comment:
why/when would this throw an exception? destructors throwing exception can in select circumstances can lead to terminate. should this be noexcept(true), or simply no such specifiers?